finetune/base.py

```diff
@@ -514,12 +520,12 @@ def _define_placeholders(self):
         self.X = tf.placeholder(tf.int32, [None, self.config.max_length, 2])  # token idxs (BPE embedding + positional)
         self.M = tf.placeholder(tf.float32, [None, self.config.max_length])  # sequence mask
         # when target dim is not set, an array of [None] targets is passed as a placeholder
-        self.Y = tf.stop_gradient(tf.placeholder(tf.float32, [None, self.target_dim or 1]))  # classification targets
+
         self.do_dropout = tf.placeholder(tf.float32)  # 1 for do dropout and 0 to not do dropout
         if self.target_type == SEQUENCE_LABELING:
             self.Y = tf.placeholder(tf.int32, [None, self.config.max_length])  # classification targets
         else:
-            self.Y = tf.placeholder(tf.float32, [None, self.target_dim])  # classification targets
+            self.Y = tf.stop_gradient(tf.placeholder(tf.float32, [None, self.target_dim or 1]))  # classification targets
```

Comment:
Gradients already do not flow into placeholders The `tf.stop_gradient` should be on the loss function to protect against changes to `self.Y`